### DISCUSSION TITLE
Running system processes from SBT tasks

### DISCUSSION THREAD

**Discussion by kincsescsaba** [sbt/sbt#7684] [Q&A]
How can I start processes from SBT tasks that require a TTY?

For example, to run a dockerized integration test, I'd need to ssh into a container, which was problematic. I tried to fix TTY-related errors (like this: [https://stackoverflow.com/questions/48648572/how-to-deal-with-pseudo-terminal-will-not-be-allocated-because-stdin-is-not-a-t](https://stackoverflow.com/questions/48648572/how-to-deal-with-pseudo-terminal-will-not-be-allocated-because-stdin-is-not-a-t)), looked for a workaround like opening a separate terminal, neither worked.

Example `build.sbt`:
```
import scala.sys.process.*

...

lazy val loginToTestAccount = taskKey[Unit]("Login to test account via SSH")
loginToTestAccount := {
  val dockerIp = s"""docker exec scalareplsecurityintegration getDockerIp""".!!.trim
  val sshCmd = s"""ssh ${integrationUser.value}@$dockerIp"""
  sshCmd.!!
}

...
```

I also tried popping up a terminal like `gnome-terminal -- bash -c "ssh user@docker-ip-address; exec bash"`.

To sum up, I'd need more information regarding how the relation of an SBT task to terminals is handled and how is it recommended to configure to achieve the desired behavior.

**Comment by eed3si9n** [comment]
```scala
import scala.sys.process.*
val p = Process("""docker exec scalareplsecurityintegration getDockerIp""")
```

then I think you can run `p.run(...)` like `p.run(ProcessLogger(line => ()), connectInput = true)`, or pass in `ProcessIO` probably?
See https://www.scala-lang.org/api/2.12.x/scala/sys/process/ProcessBuilder.html

**Reply by kincsescsaba** [reply]
I tried with the ssh command, `connectInput = true` does make a password prompt appear, though there are issues like it still does not appear to read what I type, and there is an SBT log about the time elapsed which should be hidden.
I'll try with ProcessIO
